### ISSUE TITLE
Swiss QR Bill/Invoice 30th June 2020 http://qr-bill.ch

### ISSUE THREAD

**Issue by giordano-ch** [OCA/l10n-switzerland#416]
The Swiss QR Bill will replace the payment slip.

You'll find more information under: https://www.paymentstandards.ch/dam/downloads/ig-qr-bill-en.pdf

is there already something planned? Will the QR-Invoice be a part of l10n_ch standard?

Thanks for any information.

--------------------

UPDATE

### Odoo v10 :heavy_check_mark: 

An OCA module is available at

https://github.com/OCA/l10n-switzerland/tree/10.0/l10n_ch_qr_bill

A second module to ease reconciliation process will be available soon #556 

### Odoo v11 

Currently no module is available on OCA. A forward port of Odoo v10 can be done.

### Odoo v12  :heavy_check_mark: 

Available in official `l10n_ch` module since 2020-07-16. 

### Odoo v13  :heavy_check_mark: 

Available in official `l10n_ch` module since 2020-06-26.

**Comment by yvaucher** [comment]
The QR code code should normally go in odoo core as the BVR it now maintained there.

The good place to report this can be here:

https://github.com/odoo/odoo/issues/20416

**Comment by Guisolan** [comment]
Hello, 
we (Prisme) have develop QR Bill for ODOO10
Best regards

**Comment by yvaucher** [comment]
@Guisolan welcome on Github and on the OCA.

That's great news, thanks for informing us.

Currently, Odoo Swiss users that cannot afford to migrate to Odoo 11.0 will surely need a module in former versions of Odoo. We will then probably need a version in 10.0 but also in 9.0 for former Odoo versions.

Is your module available anywhere? Github or app.odoo.com? I could find it anywhere.
Is @Prisme-Solutions-Informatique-SA willing to contribute to the Switzerland localization?

**Comment by clxadmin** [comment]
Hello,
I'm also interested to this module . @Guisolan, can you tell us more about your module? Is it available somewhere? How can we participate?

**Comment by Guisolan** [comment]
Hello,

We have develop the QR code module for Switzerland for Odoo10.

At this time we have transfer this Swiss localization to Quentin De Paoli 
<qdp@odoo.com>
- he will integrate this to the next release Odoo12 

Do you need more ?

Best regards.

_________________________________________________________


Web: www.prisme.ch
Portable: +41 79 230 54 70
Claude Guisolan
Prisme Solutions Informatique SA
Rte de Montagny 11
CH - 1772 Ponthaux
_________________________________________________________

**Comment by clxadmin** [comment]
Hello Guisolan,

I thank you for your feedback :-)

_We have develop the QR code module for Switzerland for Odoo10._
>> great. Is it somewhere available?

_Do you need more ?_
>> I also would like to run it on Odoo 11. Is this also available?

Please let me know what are the posibilities to get this module. As I wrote before I also would like to participate if there is a hand needed

Best regards

**Comment by Guisolan** [comment]
Hello,

We have send the version Odoo10 to Quentin de Paoli <qdp@odoo.com>
- I request that he install it into Odoo10 Swiss localisation
- Quentin never give us any feedback except he will us it for Odoo12
- We will make it available on Github Prisme branch today

For Odoo 11 version
- we don't plan to develop it to this version

Best regards.

_________________________________________________________


Web: www.prisme.ch
Portable: +41 79 230 54 70
Claude Guisolan
Prisme Solutions Informatique SA
Rte de Montagny 11
CH - 1772 Ponthaux
_________________________________________________________

**Comment by clxadmin** [comment]
hello,

thats very kind of you. I will watch your repo https://github.com/Prisme-Solutions-Informatique-SA/PRISME-ODOO 

We try to upgrade code for the V10 to V11 and publish it to the community if this is ok for you

**Comment by Guisolan** [comment]
Hello,

You can upgrade to v11 and publish to community.

Our v10 version was tested with https://qr-validation.iso-payments.ch 
- we recommand you do the same for v11

Best regards.
_________________________________________________________


Web: www.prisme.ch
Portable: +41 79 230 54 70
Claude Guisolan
Prisme Solutions Informatique SA
Rte de Montagny 11
CH - 1772 Ponthaux
_________________________________________________________

**Comment by clxadmin** [comment]
Hello,

I'll test the modules with https://qr-validation.iso-payments.ch :-) 

Still have no module V10 for download.  Is it possible you place it to your prism repo https://github.com/Prisme-Solutions-Informatique-SA/PRISME-ODOO ?

that would be very nice and I thank you

**Comment by giordano-ch** [comment]
@clxadmin I think you have to change the branch to 10.0 then you'll find it (prisme_invoice_iban_qrcode)

**Comment by Guisolan** [comment]
Hello,

My collegue told me that our dev is online into branch 10.0 (you have to 
select this branch to reach it).
Direct access is: 
https://github.com/Prisme-Solutions-Informatique-SA/PRISME-ODOO/tree/10.0/prisme_invoice_iban_qrcode 


Best regards.

_________________________________________________________


Web: www.prisme.ch
Portable: +41 79 230 54 70
Claude Guisolan
Prisme Solutions Informatique SA
Rte de Montagny 11
CH - 1772 Ponthaux
_________________________________________________________

**Comment by clxadmin** [comment]
I finally found the module -- I thank you

**Comment by yvaucher** [comment]
@Guisolan Thank you for the link!

You might want to specify the licence in the module `__manifest__.py` file. Which should be AGPL as the modules depends on `l10n_ch_payment_slip`.

Talking about `l10n_ch_payment_slip`
For v11, I must state that I first thought the official ISR report in Odoo core would be enough, but it doesn't cover all needs.

Thus, at first we abandoned the idea of porting `l10n_ch_payment_slip` to v 11.
However, lastly we reopened https://github.com/OCA/l10n-switzerland/pull/373 to complete it.

And as PRISM module depends on it, I'm glad we did, this should ease the port of PRISME modue from v10 to v11.

However, it seems that the PRISME module for QR code doesn't need all features of `l10n_ch_payment_slip`. It could be a better approach to extract the current reportlab ISR report into a second module.

In the long term, QR code printout should be included in Odoo core. But, it usually takes time for Odoo to include such features. And users will need it in previous versions too only few will start right away on v 12. Thus, if there is any PR for v11 to include it in this repository just ping me and I'll review it. Even a backport to v9 could be possible.

**Comment by robertrottermann** [comment]
We are in the process to create a report for the iso20022 qrcode invoice. It will be for V9.
What we did so far is based on the v9 Paymentslip module
Tomorrow I will discuss with my team how to proceed.

**Comment by robertrottermann** [comment]
According to SIX the QR code will only be mandatory  June 30th 2020
So there is no immediate hurry to provide it.
This Info I just got by a customer that is very close to SIX.

**Comment by pegonzalezs** [comment]
Dear @robertrottermann @yvaucher 
my company needs this functionality for v9, so I would like to contribute on this backport. So, we keep us posted about it.
Thank you

**Comment by Guisolan** [comment]
Salut David,
Est-ce que tu comprends cette demande?
Ciao et merci pour ton feedback

Envoyé depuis IBM Verse


   Pedro Evaristo Gonzalez Sanchez --- Re: [OCA/l10n-switzerland] Swiss QR Bill/Invoice 2019 (#416) --- 
    De :"Pedro Evaristo Gonzalez Sanchez" <notifications@github.com>A :"OCA/l10n-switzerland" <l10n-switzerland@noreply.github.com>Cc :"Guisolan" <cg@prisme.ch>, "Mention" <mention@noreply.github.com>Date :mar., 24 juil. 2018 10:19Objet :Re: [OCA/l10n-switzerland] Swiss QR Bill/Invoice 2019 (#416)
  
  Dear @robertrottermann @yvaucher my company needs this functionality for v9, so I would like to contribute on this backport. So, we keep us posted about it. Thank you

**Comment by didierdonze** [comment]
Hello,
Any updates in this thread? Did someone open a PR to migrate the Prisme module to V11?

**Comment by Guisolan** [comment]
Hello,

Unfortunatelly no for short time.

We plan to migrate all Prisme dev from v10 to v11 and from v11 to 12 in Q4 
2019.

Reason
- we plan to hire 1 ressource in september 2019
- we estimate that in Q4-2019 we can deliver

Best regards.
_________________________________________________________


Web: www.prisme.ch
Portable: +41 79 230 54 70
Claude Guisolan
Prisme Solutions Informatique SA
Rte de Montagny 11
CH - 1772 Ponthaux
_________________________________________________________

**Comment by didierdonze** [comment]
Hi,
Some news on the subject. Odoo did developp the feature in V12:
https://github.com/odoo/odoo/commit/68dafa0620fb8e36c9a1a58da0668bede2c3699b
It's hidden for the moment, but can be teste
![image](https://user-images.githubusercontent.com/5780150/52045806-8d978780-2545-11e9-8a4c-b18d7574b607.png)
d.

The slip is missing for the moment. I guess no need to develop a community module in V12, for the moment at least, we might need to backport to V10/11 for our customers.

**Comment by Guisolan** [comment]
Hello,

Yes it is done with ODOO12 because 
- we have send the code from our ODOO 10 development to Quentin De Paoli 
(who is the ODOO product manager from accounting and localisation)
- at this time Quentin never ask us to help him to check his integration

Best regards.
_________________________________________________________


Web: www.prisme.ch
Portable: +41 79 230 54 70
Claude Guisolan
Prisme Solutions Informatique SA
Rte de Montagny 11
CH - 1772 Ponthaux
_________________________________________________________

**Comment by clxadmin** [comment]
Hello,

Yes, looks like its only available for 12 Enterprise
![Screenshot - 2019-03-14T143134 986](https://user-images.githubusercontent.com/9445285/54360846-11c14c80-4666-11e9-94a0-4b20071a84cf.png)

**Comment by yvaucher** [comment]
@clxadmin the option in enterprise is **not** the Swiss QR Code

SEPA QR Code != Swiss QR code invoice

Currently the report is in Odoo Community core

You can see the xml file that defines the addition to the invoice report here:
https://github.com/odoo/odoo/blob/12.0/addons/l10n_ch/report/swissqr_report.xml

You can activate it by adding an `ir.config_parameter` named : `l10n_ch.print_qrcode`

**Comment by yvaucher** [comment]
Some news about QR-Invoice here is a PR on master which should improve it's launching on 20th of June 2020.

https://github.com/odoo/odoo/pull/35931

**Comment by yvaucher** [comment]
Changed the date in the title for QR Invoice launch according to http://qr-bill.ch

**Comment by oco-odoo** [comment]
I'm currently refactoring QR codes in master (so, Swiss ones as well). In the meantime, I'll update the implementation to match the latest version of the specification. Once it's done, I'll do the same in stable releases. 

I'll probably have a few questions in the upcoming days, so that were sure to cover all the different use cases, so stay tuned ;)

**Comment by yvaucher** [comment]
This one deserved to be pinned

**Comment by oco-odoo** [comment]
@yvaucher You don't say! (and your timing is impeccable :) )

I have finished refactoring the QR codes. It's still only the first version of the code and hasn't been reviewed in any way yet, but you can already test the Swiss part of it if you want :) Branch is called master-qr-code-refact-oco ; it is accessible on runbot as well. 


Community PR: https://github.com/odoo/odoo/pull/44839 
Enterprise PR: https://github.com/odoo/enterprise/pull/8262

**Comment by stevech091** [comment]
Hello,
did someone done a V8 version of QR invoices module?
Thanks

**Comment by SimoRubi** [comment]
Hi all,
first: BIG thanks for all your work in swiss localization.
I see (https://github.com/OCA/l10n-switzerland/issues/416#issuecomment-583337874) you implemented this change in `master` branch for Odoo so they will make it available for supported versions, but what about unsupported versions?

For instance:
- v10: prisme's module has been removed by https://github.com/Prisme-Solutions-Informatique-SA/PRISME-ODOO/commit/b09d19acf37a281f613d0202edcb91cd6b2f0087, is there any module available right now?
- v8: is there any work in progress?

**Comment by yvaucher** [comment]
@stevech091 @SimoRubi for older version there is no rush to replace ISR by QR-Bill, the ISR will be still supported till 2022. Both will be accepted in parallel.

I don't know of any module that does it for older Odoo versions to generate invoices.

As QR-Bill are basically an IBAN (QR IBAN) and a structured reference (QRR). You should also be able to already receive such invoices.

QR-Bills are much nicer though, so I understand if you want them as soon as possible in older Odoo versions. If you want this, the to path follow might be to backport Odoo's report into previous versions. It's some work as some part of the code is hidden in l10n_ch module and needs to be extracted in an other module.

**Comment by davidwul** [comment]
From what I see, all of this is for outgoing payment. (customer bills)
Regarding in payment (vendor bills): using pain001 (with l10n_ch_pain_credit_transfer). It seems this is not yet ready.
Is odoo SA willing to make this work? Or should we provide the necessary changes (in the l10n_ch_pain_credit_transfer OCA module) 
Any idea, or suggestion?

**Comment by pegonzalezs** [comment]
Hi @davidwul,

thanks for your question. Me and Robert Rotterman are waiting to have the
dependencies already finished to continue our work.

Maybe Yannick can provide us some feedback to know if we need to wait till
l10n_ch_base_bank is finished and also about other suggestions or
recommendations

Greetings
Pedro

**Comment by davidwul** [comment]
Do you mean, changes should be done by Odoo and will end up in l10n_ch_base_bank ?
Will this be done for 13 and for 12 ?

**Comment by yvaucher** [comment]
I spend some time this week reviewing the official QR invoice in V13.

Here are my findings and fixes:

https://github.com/odoo/odoo/pull/53084


Before:

![qr-bill-anotations](https://user-images.githubusercontent.com/4158438/84823391-8976e780-b01e-11ea-86f9-33db28cc4624.png)

After:

![2020-06-16-212105_1099x559_scrot](https://user-images.githubusercontent.com/4158438/84823412-9398e600-b01e-11ea-9dd3-c2cbcba2ae4f.png)

**Comment by yvaucher** [comment]
@davidwul SEPA is a bit out of scope of this issue. So in short Odoo has an enterprise addon for it, they won't contribute to the OCA module l10n_ch_pain_credit_transfer. Their enterprise addon `account_sepa` is not fully implemented for Swiss SEPA nevertheless. You might create a new issue about it for more details.

**Comment by yvaucher** [comment]
@pegonzalezs The aim is to integrate all `l10n_ch_base_bank` vital features into offical `l10n_ch`. Please see this mail on the mailing list: https://odoo-community.org/groups/switzerland-29/switzerland-162989

**Comment by yvaucher** [comment]
Here is a backport for version 10.0 of the QR-bill based on the one made by Odoo

https://github.com/OCA/l10n-switzerland/pull/552

**Comment by yvaucher** [comment]
New update:

1. Fixes for the official QR-bill report are about to be merged in the core: https://github.com/odoo/odoo/pull/53084
2. Here you will find a new module PR to import QR-bill with a PayEye scanner. #553 It will also support importing PDF files soon.

**Comment by yvaucher** [comment]
New update on Swiss QR-bill generation:

For v10: https://github.com/OCA/l10n-switzerland/tree/10.0/l10n_ch_qr_bill
For v11 it needs to be ported and hosted in OCA/l10n-switzerland (from v10 as it already makes the module separately)
For v12 we have to wait on Odoo to finalize it: odoo/odoo#54427
For v13 it is available in `l10n_ch` with 2 ongoing fixes for margin issues https://github.com/odoo/odoo/pull/54383

**Comment by yvaucher** [comment]
@robertrottermann you mentioned the version 9.0 right? any news about it? Do you want to go with your version or try to backport the v10 backport?

@Guisolan about the version for 10.0, you mentioned it 2 years ago (forgot about it and now I assume you didn't wanted to contribute to the OCA), with the time I spent fixing the official one I choose to backport the one from Odoo, I honestly don't know if Quentin De Paoli based their work on yours. What I know is that their first version ~2 years ago was the QR bill version 1.x and now it has changed a lot to apply the specs changes for v 2.1 .

@ all Feed backs would be welcome on the QR bill generation.

BTW as the only active user with merge access in this PSC, I'm really not against having some other PSC members that want to contribute. Especially if you think I'm taking arbitrary decision that I shouldn't. Just raise your hand if you have some contribs in that repository already :) At least this thread shows we are not a few around this localization.

**Comment by Guisolan** [comment]
Hello Yannick,

Regarding Swiss QR Bill finally we have develop it at v10, v11, v12.
Our dev was validated with La Poste and UBS.
We manage the reference field (27 char).

This dev is available at apps.odoo.com (search under qr-bill).
We sold you dev because we have develop each ODOO version (10, 11, 12).

We don't have develop v13 because QR-Bill is deliver within l10n_ch.

We know 2 problems with this version:
- bank reference field is not automatically updated (we have develop shot 
module to do it) meanwhile that OCA enhance this point
- reference (27 char) is not printed into the qr-bill we don't know if you 
indent to do it during the period on which ISR is still available

You can phone me if you want addl info (in french would it be more easy).

Best regards.

_________________________________________________________


Web: www.prisme.ch
Portable: +41 79 230 54 70
Claude Guisolan
Prisme Solutions Informatique SA
Rte de Montagny 11
CH - 1772 Ponthaux
_________________________________________________________

**Comment by Martronic-SA** [comment]
@Guisolan 
>  bank reference field is not automatically updated (we have develop shot
module to do it) meanwhile that OCA enhance this point

Is that related to that issue? https://github.com/OCA/bank-payment/issues/720

**Comment by Guisolan** [comment]
Hello,

Our module build automatically the bank reference field when customer 
invoice is validated.

Best regards.
_________________________________________________________


Web: www.prisme.ch
Portable: +41 79 230 54 70
Claude Guisolan
Prisme Solutions Informatique SA
Rte de Montagny 11
CH - 1772 Ponthaux
_________________________________________________________

**Comment by yvaucher** [comment]
@Guisolan 

> We don't have develop v13 because QR-Bill is deliver within l10n_ch.

Well sort of, I'm well aware of this as I gave insights and proposed patches to Odoo team to fix it in v13 and spent hours on this. As explained in previous comments. Now It's pretty decent. Will see, soon, if there are still few glitches.

Be aware that Odoo is backporting the report to v12 in `l10n_ch`. Should work for your module if you keep the official report hidden. (should still be hidden by default)

About the 27 char ref, also known as QRR, it must only be printed in Reference field if it is associated with a QR-IBAN, otherwise it can sort of be passed in "Additional info" as it won't be considered as a structured reference.


> bank reference field is not automatically updated (we have develop shot
> module to do it) meanwhile that OCA enhance this point
I don't understand this point. I believe you are talking about the the v13 official module? How is it related to the OCA?
If not already done could you report the issue somewhere?

**Comment by SimoRubi** [comment]
> New update on Swiss QR-bill generation:
> 
> For v10: https://github.com/OCA/l10n-switzerland/tree/10.0/l10n_ch_qr_bill
> For v11 it needs to be ported and hosted in OCA/l10n-switzerland (from v10 as it already makes the module separately)
> For v12 we have to wait on Odoo to finalize it: [odoo/odoo#54427](https://github.com/odoo/odoo/pull/54427)
> For v13 it is available in `l10n_ch` with 2 ongoing fixes for margin issues [odoo/odoo#54383](https://github.com/odoo/odoo/pull/54383)

Update for v12: https://github.com/odoo/odoo/pull/54427 is merged :tada:

**Comment by thomich** [comment]
testing around on Odoo 10 with l10n_ch_qr_bill:

duplicate partner_bank_id in invoice/other info:
https://github.com/OCA/l10n-switzerland/blob/10.0/l10n_ch_qr_bill/views/account_invoice_view.xml#L16

results in:
![image](https://user-images.githubusercontent.com/4398840/88283833-842b4d80-ccec-11ea-807b-b816f7271346.png)

**Comment by yvaucher** [comment]
@thomich AFAIK the field `partner_bank_id` is invisible in Odoo CE. So we have to change that to make it visible instead of adding the field a second time. In your case you see twice the field maybe because an other module already makes the field visible?

In any case you are right it's a bug even with an invisible field.

https://github.com/odoo/odoo/blob/10.0/addons/account/views/account_invoice_view.xml#L465

**Comment by yvaucher** [comment]
@thomich review welcome on https://github.com/OCA/l10n-switzerland/pull/555

It should fix the partner bank field issue.

**Comment by thomich** [comment]
> @thomich review welcome on #555
> 
> It should fix the partner bank field issue.

hey @yvaucher 
thanks... looks great

another question... 
it will generate a new QRR (reference) which is the uid used for the payment of the customer, right?
with ISR (BVR) we also have a ref which is in bank camt file we import to reconcile invoice moves.
has the camt import method also changed as we have new QRR and field for QRR is l10n_ch_qrr / l10n_ch_qrr_spaced.

Thanks for clearify.

Regards
thomi

**Comment by yvaucher** [comment]
@thomich yes the QRR becomes the new payment reference. It's intentionally the same format as the ISR (27 chars).

Reconciliation must be done on it. It should already work on v12 and v13. The point you raise looks to me that in v10 we probably still need to copy it on the transaction_id field. Which means we are missing an additional module.

BTW, be aware that a QR-IBAN is usually an alias for an IBAN account. Thus for reconciliation you might need to reconcile the statement of the IBAN on the QR-IBAN. I haven't seen a camt about it, yet so I don't know if we will have to tweak something more if you use both the QR-IBAN and the IBAN of your bank account. I expect it to work without issue if you use this bank account only with QR-IBAN, until I'm proven wrong. As Odoo doesn't support any kind of alias bank account for the same account.

**Comment by thomich** [comment]
hey @yvaucher 

thanks for reply...

> @thomich yes the QRR becomes the new payment reference. It's intentionally the same format as the ISR (27 chars).

:+1: 

> Reconciliation must be done on it. It should already work on v12 and v13. The point you raise looks to me that in v10 we probably still need to copy it on the transaction_id field. Which means we are missing an additional module.

means that in v10 special method will be made to copy QRR to transaction_id?

> BTW, be aware that a QR-IBAN is usually an alias for an IBAN account. Thus for reconciliation you might need to reconcile the statement of the IBAN on the QR-IBAN. I haven't seen a camt about it, yet so I don't know if we will have to tweak something more if you use both the QR-IBAN and the IBAN of your bank account. I expect it to work without issue if you use this bank account only with QR-IBAN, until I'm proven wrong. As Odoo doesn't support any kind of alias bank account for the same account.

yeah... it is really confusing... also for our customers...


one other thing i found:
if you print the QR-Invoice, it will not give any file extension, which can cause problems on some client systems.
it's a PDF, but with out .pdf extension..

regards
thomi

**Comment by yvaucher** [comment]
@thomich Thank you for the feed back.

Here are 2 PRs that should resolves the following issues:

- reconciliation with `base_transaction_id` #556 
- pdf file extension #557

**Comment by thomich** [comment]
hey @yvaucher 

man thanks for your work... tested and looks very nice...
have made the german translation... i know, i could make a PR and so on, but think if you integrate it, it's faster.. ;)

pm me please, then i can send you the de.po file...

again, many thanks
thomi

**Comment by stevech091** [comment]
Hello,
I have made italian translation on Weblate.
Regards

**Comment by yvaucher** [comment]
@thomich the proper way to push translation on OCA modules is through Weblate.

https://translation.odoo-community.org/projects/l10n-switzerland-10-0/

We have a bot that will automate the push on github.
IIRC Weblate suports import a PO file, but it seems deactivated, maybe a translator with enough admin rights can do it if you have numerous terms to translate.

**Comment by yvaucher** [comment]
@stevech091 Great! Grazie !

**Comment by thomich** [comment]
> @thomich the proper way to push translation on OCA modules is through Weblate.
> 
> https://translation.odoo-community.org/projects/l10n-switzerland-10-0/
> 
> We have a bot that will automate the push on github.
> IIRC Weblate suports import a PO file, but it seems deactivated, maybe a translator with enough admin rights can do it if you have numerous terms to translate.

Hey @yvaucher 

thx, will do that.... will translations then also be ported to other version's? 11, 12, 13?

regards
thomi

**Comment by yvaucher** [comment]
@thomich for the QR-bill for 12 and 13 it doesn't exists in OCA, it's in odoo CE

You want to have a look at https://github.com/odoo/odoo/wiki/Translations

**Comment by thomich** [comment]
@yvaucher what about QR-bill and 11.0?

thanks for information about translation, will make it soon ;)

thomi

**Comment by thomich** [comment]
> @thomich the proper way to push translation on OCA modules is through Weblate.
> 
> https://translation.odoo-community.org/projects/l10n-switzerland-10-0/
> 
> We have a bot that will automate the push on github.
> IIRC Weblate suports import a PO file, but it seems deactivated, maybe a translator with enough admin rights can do it if you have numerous terms to translate.

Hey @yvaucher 
have registered @ weblate but can't make translations or even suggest. I also contacted transbot@odoo-community.org but not getting any notification/feedback.

Here is my de.po file.... if you can do it with a PR... would be nice.. and THANKS
[de.tar.gz](https://github.com/OCA/l10n-switzerland/files/5101626/de.tar.gz)

regards
thomi

**Comment by philhug** [comment]
Is anyone working on adding  Creditor Reference (SCOR) support instead of QRR?
It uses the normal IBAN account number + a standard ISO 11649 reference (starting with RF...) instead of a separate QR-IBAN

https://www.paymentstandards.ch/dam/downloads/ig-qr-bill-en.pdf

**Comment by yvaucher** [comment]
@philhug the official odoo module does not support Creditor Reference. And I strongly doubt that Odoo will do it any time soon.

On my side I have no ongoing implementation on this topic.
Shouldn't be too complicated to add though. We have to propose that to the official module. In hope they don't only accept it in version 14.0.

**Comment by yvaucher** [comment]
FYI https://github.com/odoo/odoo/pull/55571 has been merged.

It includes a new module `l10n_ch_qriban` which adds a field on `res.partner.bank` to store the QR-IBAN.
This aims to allows reconciliation using both the IBAN and the QR-IBAN with the in-payments.

cc @thomich

**Comment by R3D2** [comment]
Hey ! I tried to migrate the module to V11, it seems to work fine. I'm really new to contributing, I will be glad to have my code reviewed. https://github.com/R3D2/l10n_ch_qr_bill

**Comment by thomich** [comment]
> Hey ! I tried to migrate the module to V11, it seems to work fine. I'm really new to contributing, I will be glad to have my code reviewed. https://github.com/R3D2/l10n_ch_qr_bill

Hey @R3D2 

Thanks for your contribution.

Check license, as i know, it should keep the original license from Odoo 13: LGPL-3
Check if license is everywhere the same ;)

Regards
thomi

**Comment by yvaucher** [comment]
@R3D2 can you push it as a pull request in this repository on branch 11.0 to allow us to review it ?

**Comment by gfrino** [comment]
We at ticinoWEB have developed module for Odoo v10 https://apps.odoo.com/apps/modules/10.0/invoice_ch_qr_slip/ and a fix for Odoo QR Bill for v14 https://apps.odoo.com/apps/modules/14.0/generate_inv_qr/

The v14 fixes this error: The following error prevented 'Swiss QR bill' QR-code to be generated though it was detected as eligible: When using a QR-IBAN as the destination account of a QR-code, the payment reference must be a QR-reference.

Our module will automatically generate the payment reference
